### ISSUE TITLE
Listbox should not require a name when its used in a combobox pattern

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3860]
This example should not fail. It comes straight out of WAI-ARIA 1.2:

https://www.w3.org/TR/wai-aria-1.2/#combobox

```html
<label for="tag_combo">Tag</label>
<input type="text" id="tag_combo"
      role="combobox" aria-autocomplete="list"
      aria-haspopup="listbox" aria-expanded="true"
      aria-controls="popup_listbox" aria-activedescendant="selected_option">
<ul role="listbox" id="popup_listbox">
   <li role="option">Zebra</li>
   <li role="option" id="selected_option">Zoom</li>
</ul>
```

I think the fix would be to skip listbox if its part of a combobox pattern in the `no-naming-method-matches` method.

There's an issue open on WAI-ARIA for it: https://github.com/w3c/aria/issues/1769#issuecomment-1386857482

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base,  with the "role=listbox" , it is not needed any name when it is used in a combobox. No `aria` rules failing
Test snippet:
```
<label for="tag_combo">Tag</label>
<input type="text" id="tag_combo"
      role="combobox" aria-autocomplete="list"
      aria-haspopup="listbox" aria-expanded="true"
      aria-controls="popup_listbox" aria-activedescendant="selected_option">
<ul role="listbox" id="popup_listbox">
   <li role="option">Zebra</li>
   <li role="option" id="selected_option">Zoom</li>
</ul>
```


![Image](https://user-images.githubusercontent.com/47409579/217759746-499717a2-cff8-4481-b2b2-23aafaa75398.png)
